### PR TITLE
 [API-Design-Guidelines]: fix typos and internal references, format tables, restructure sentences

### LINKED ISSUE #232: fixing typos and other style-issues in documentation-files
**Problem description**
Currently the documentation-files, e.g., API-design-guideline.md has some broken internal references (e.g., inside the table of content).
Also some other issues were found like: 

- unformated tables
- too long sentences, which were hardly to read on the first try.
- missing articles
- some grammar issues

### PR DESCRIPTION
**Pull request by maxl2287**
#### What type of PR is this?

Add one of the following kinds:

* correction

#### What this PR does / why we need it:
This PR is a fixing of some style and reference-issues for the API design guidelines

#### Which issue(s) this PR fixes:
Fixes #232

### CODE REVIEW

**Review by patrice-conil — DISMISSED** [review]
/LGTM
Thanks @maxl2287  for this great job

File: documentation/API-design-guidelines.md [hunk 1]
```diff
@@ -1417,64 +1429,64 @@ Notes:
...
+|----------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| ACTIVATION_REQUESTED | Subscription creation (POST) is triggered but subscription creation process is not finished yet.                                                                                                                                                                      |
+| ACTIVE               | Subscription creation process is completed. Subscription is fully operative.                                                                                                                                                                                          |
+| DEACTIVE             | Subscription is temporarily inactive, but its workflow logic is not deleted. DEACTIVE could be used when an user initially provided consent for the event monitor and then later denied this consent. For now we did not provide capability to reactive subscription. |
```
Comment (PedroDiez) [inline documentation/API-design-guidelines.md:1480]:
```suggestion
| INACTIVE             | Subscription is temporarily inactive, but its workflow logic is not deleted. DEACTIVE could be used when an user initially provided consent for the event monitor and then later denied this consent. For now we did not provide capability to reactive subscription. |
```

Comment (PedroDiez) [reply documentation/API-design-guidelines.md:1480]: to align with template value

**Review by PedroDiez — COMMENTED** [review]
Minor editorial comments. Great work!

File: documentation/API-design-guidelines.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ This document captures guidelines for the API design in CAMARA project. These gu
   - [2. APIfication Principles](#2-apification-principles)
     - [2.1 Domain Driven Design](#21-domain-driven-design)
     - [2.2 API First](#22-api-first)
-    - [2.3 Interface standardization. Standardization fora.](#23-interface-standardization-standardization-fora)
+    - [2.3 Interface standardization. Standardization fora.](#23-interface-standardization-standardization-fora-)
```
[the remark this answers is not in the document]
Comment (maxl2287) [reply documentation/API-design-guidelines.md:14]: @PedroDiez thanks for reporting.

Weird behaviour again, that this is working with my local IDE (IntelliJ).
I will take a look on that :)

image

[the remark this answers is not in the document]
Comment (maxl2287) [reply documentation/API-design-guidelines.md]: Okay, main issue was an additional whitespace after the headline.
I removed the whitespace and reverted it back to the references without a dash at the end.

**Comment by maxl2287** [comment]
@patrice-conil & @PedroDiez updates are done.

**Comment by PedroDiez** [comment]
Taking advantage in Table contents Max if you can fix this

    - [### 6.1 Standardized use of CAMARA error responses](#61-standardized-use-of-camara-error-responses)
    - [### 6.2 Error Responses - Device Object](#62-error-responses---device-object)

**Comment by maxl2287** [comment]
@PedroDiez done ✅

**Review by patrice-conil — APPROVED** [review]
LGTM

**Review by PedroDiez — APPROVED** [review]
LGTM

**Review by shilpa-padgaonkar — DISMISSED** [review]
/LGTM

File: documentation/API-design-guidelines.md [hunk 3]
```diff
@@ -634,67 +664,67 @@ In the following, we elaborate on the existing client errors. In particular, we
...
-| 2 | Some identifier cannot be matched to a device | 404 | DEVICE_NOT_FOUND | Device identifier not found. |  
-| 3 | Device identifiers mismatch | 422 | DEVICE_IDENTIFIERS_MISMATCH | Provided device identifiers are not consistent. |
-| 4 | Invalid access token context | 403 | INVALID_TOKEN_CONTEXT | Device identifiers are not consistent with access token. |
-| 5 | Service not applicable to the device | 422 | DEVICE_NOT_APPLICABLE | The service is not available for the provided device. |
+| **Case #** | **Description**                                                            | **Error status** |         **Error code**         | **Message example**                                      |
+|:----------:|:---------------------------------------------------------------------------|:----------------:|:------------------------------:|:---------------------------------------------------------|
+|     0      | The request body does not comply with the schema                           |       400        |        INVALID_ARGUMENT        | device does not comply with the schema.                  |
```
Comment (Kevsy) [inline documentation/API-design-guidelines.md:722]:
```suggestion
|     0      | The request body does not comply with the schema                           |       400        |        INVALID_ARGUMENT        | Request body does not comply with the schema.                  |
```

**Review by Kevsy — APPROVED** [review]

**Review by rartych — APPROVED** [review]

**Review by PedroDiez — APPROVED** [review]
LGTM

**Review by rartych — APPROVED** [review]

**Review by PedroDiez — APPROVED** [review]
LGTM

**Comment by rartych** [comment]
Merging as commonly approved (the last commits resolved conficts)

**Pull request merged**